src/App.js

```diff
@@ -1,30 +1,39 @@
-import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import React, { useState } from 'react';
+import { Route, Switch, Redirect } from 'react-router-dom';
 import './App.css';
-import Test from './components/Test';
 import Navigation from './components/Navigation';
 import AddItem from './components/AddItem';
 import List from './components/List';
 import Welcome from './components/Welcome';
+import getToken from './lib/tokens';
+// import TestWelcome from './components/TestWelcome';
+// import CreateNewList from './components/CreateNewList';
 
 function App() {
+  const [token, setToken] = useState(localStorage.getItem('userToken') || '');
+
+  const handleClick = () => {
+    const token = getToken();
+    localStorage.setItem('userToken', token);
+    setToken(token);
+  };
+
   return (
     <div className="App">
       <Switch>
-        <Route path="/list">
-          <List />
-        </Route>
+        <Route path="/list">{token ? <List /> : <Redirect to="/" />}</Route>
 
         <Route path="/add-item">
-          <AddItem />
+          {token ? <AddItem /> : <Redirect to="/" />}
         </Route>
 
         <Route path="/">
-          <Welcome />
+          {token ? <Redirect to="/list" /> : <Welcome onClick={handleClick} />}
+          {/* <TestWelcome />
+          {token ? <Redirect to="/list" /> : <CreateNewList onClick={handleClick} />} */}
```

Comment:
Same here, can get rid of the commented out code.

Reply:
👍